src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,507 @@ title: Builder.io & Astro
 description: Add content to your Astro project using Builder.io’s visual CMS
 type: cms
 service: Builder.io
-stub: true
+stub: false
 i18nReady: false
 ---
 import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import FileTree from '~/components/FileTree.astro';
 
+[Builder.io](https://www.builder.io/) is a visual CMS that supports drag and drop content editing for building web sites.
 
+At the end of this recipe you'll be able to create blog posts in the Builder app and have your Astro site update with this new content automatically using a webhook.
+
+To do this, you'll use Builder's [Content API](https://www.builder.io/c/docs/content-api) and [Qwik API](https://www.builder.io/c/docs/qwik-api) to connect your Builder space to Astro with zero client-side JavaScript.
+
+:::tip[Official resource]
+If you are using a JavaScript framework (e.g. Svelte, Vue, or React) in your Astro project you can use one of Builder's integrations as an alternative to making raw fetch calls through the REST API.
+:::
+
+## Prerequisites
+
+To get started, you will need to have the following:
+
+* **A Builder account and space** - If you don't have an account yet, [sign up for free](https://www.builder.io/) and create a new space. If you already have a space with Builder, feel free to use it, but you will need to modify the code to match the model name and custom data fields.
+* **A Builder API key** - This public key will be used to fetch your content from Builder. [Read Builder's guide on how your key](https://www.builder.io/c/docs/using-your-api-key#finding-your-public-api-key).
+
+## Setting up credentials
+
+To add your Builder API key to Astro, create a `.env` file in the root of your project with the following variable:
+
+```ini title=".env"
+BUILDER_API_PUBLIC_KEY=YOUR_API_KEY
+```
+
+Now, you should be able to use this API key in your project.
+
+:::note
+At the time of writing, [this key is public](https://www.builder.io/c/docs/using-your-api-key#prerequisites), so you don't have to worry about hiding or encrypting it.
+:::
+
+If you would like to have IntelliSense for your environment variables, you can create a `env.d.ts` file in the `src/` directory and configure `ImportMetaEnv` like this:
+
+```ts title="src/env.d.ts"
+interface ImportMetaEnv {
+  readonly BUILDER_API_PUBLIC_KEY: string;
+}
+```
+
+Your project should now include these new files:
+
+<FileTree title="Project Structure">
+- src/
+  - **env.d.ts**
+- **.env**
+- astro.config.mjs
+- package.json
+</FileTree>
+
+
+## Making a blog with Astro and Builder
+
+With the set up above, you can now create a blog with Astro and Builder.
+
+### Creating a model for a blog post
```

Comment:
This section is a lot about using Builder, and we typically like to avoid getting into too many of the third-party's instructions.

Sometimes, we'd offload an entire section like this and put it in prerequisites e.g. you need a Builder project with a blog post model. This is because for example, Builder dashboard menus could change, and we really wouldn't have a good way of knowing until someone says these instrutions don't work for them.  😅 

Can we actually cut this section, and just say you're going to need this set up in the prerequisites section and try to stick to what Astro is going to need and what it's important to do have to pick it up at the point where you're integrating with Astro? In the earlier section when you create the `.env` file then, you can ALSO add the modal name in that step from your existing project. You can use something like: "The example below shows creating the preview for a modal named `blogpost`." This is how we get around the fact that we might not know what they called it, but we're going to assume this name...

Reply:
I tend to agree. The instructions relate to another site whose UI is liable to change. Builder has an interest to give clear and up-to-date instructions, so we can just link to them. Keeps the guide shorter visually, too.

BTW I wrote this section following other CMS recipes. I suggest we make this policy clear to future contributors, so they can save themselves some work. Put it in the relevant writing guide(s). Let me know what You think, I'd be happy to open a PR!

Reply:
I feel I was slightly involved in this migration :o). I've set a reminder and will check back with y'all on this in a few weeks.